src/python/pants/option/options_test.py

```diff
@@ -853,171 +869,180 @@ def test_enum_option_type_parse_error(self):
+      expected='crufty1',
+    )
+    assert_deprecation_triggered(
+      './pants',
+      config={'stale': {'crufty': 'stale_and_crufty'}},
+      scope='stale',
+      option='crufty',
+      expected='stale_and_crufty',
+    )
 
     # Make sure the warnings don't come out for regular options.
     with self.warnings_catcher() as w:
       self._parse('./pants stale --pants-foo stale --still-good')
       self.assertEqual(0, len(w))
 
-  def test_deprecated_options_env(self):
-    with self.warnings_catcher() as w:
-      options = self._parse('./pants', env={'PANTS_GLOBAL_CRUFTY':'crufty1'})
-      self.assertEqual('crufty1', options.for_global_scope().global_crufty)
-      self.assertOptionWarning(w, 'global_crufty')
-
-    with self.warnings_catcher() as w:
-      options = self._parse('./pants', env={'PANTS_STALE_CRUFTY':'stale_and_crufty'})
-      self.assertEqual('stale_and_crufty', options.for_scope('stale').crufty)
-      self.assertOptionWarning(w, 'crufty')
-
-  def test_deprecated_options_config(self):
-    with self.warnings_catcher() as w:
-      options = self._parse('./pants', config={'GLOBAL':{'global_crufty':'crufty1'}})
-      self.assertEqual('crufty1', options.for_global_scope().global_crufty)
-      self.assertOptionWarning(w, 'global_crufty')
-
-    with self.warnings_catcher() as w:
-      options = self._parse('./pants', config={'stale':{'crufty':'stale_and_crufty'}})
-      self.assertEqual('stale_and_crufty', options.for_scope('stale').crufty)
-      self.assertOptionWarning(w, 'crufty')
-
   @unittest.mock.patch('pants.base.deprecated.PANTS_SEMVER', Version(_FAKE_CUR_VERSION))
-  def test_delayed_deprecated_option(self):
+  def test_delayed_deprecated_option(self) -> None:
     with self.warnings_catcher() as w:
-      delayed_deprecation_option_value = (
-        self._parse(
-          './pants --global-delayed-deprecated-option=xxx')
-        .for_global_scope()
-        .global_delayed_deprecated_option)
+      delayed_deprecation_option_value = self._parse(
+          './pants --global-delayed-deprecated-option=xxx'
+        ).for_global_scope().global_delayed_deprecated_option
       self.assertEqual(delayed_deprecation_option_value, 'xxx')
       self.assertEqual(0, len(w))
 
     with self.warnings_catcher() as w:
-      delayed_passed_option_value = (
-        self._parse(
-          './pants --global-delayed-but-already-passed-deprecated-option=xxx')
-        .for_global_scope()
-        .global_delayed_but_already_passed_deprecated_option)
+      delayed_passed_option_value = self._parse(
+        './pants --global-delayed-but-already-passed-deprecated-option=xxx'
+      ).for_global_scope().global_delayed_but_already_passed_deprecated_option
       self.assertEqual(delayed_passed_option_value, 'xxx')
       self.assertOptionWarning(w, 'global_delayed_but_already_passed_deprecated_option')
 
-  def test_mutually_exclusive_options_flags(self):
+  def test_mutually_exclusive_options(self) -> None:
```

Comment:
Same as with the deprecation test, I combined all the `mutuallY_exclusive_options` tests into one single test and used two helper functions to reduce duplication.